utilities/serviceutils.py

```diff
@@ -1,20 +1,19 @@
 from config import Services
+from dataclasses import dataclass, field
 from datetime import datetime, timezone
 from typing import Optional
-from enum import Enum
-
 
+@dataclass
 class ServiceRole:
-    def __init__(self, name: str, id: str):
-        self.name = name
-        self.id = id
+    name: str
+    id: str
+    _role: object = field(default=None, init=False)
 
     def __eq__(self, other: object) -> bool:
         if isinstance(other, int):
             return self.id == other
-        if hasattr(self, "_role"):
-            if type(self._role) == type(other):
-                return self._role == other
+        if self._role is not None and type(self._role) == type(other):
+            return self._role == other
```

Comment:
why use `field` with `init=False` when changing not to use `hasattr`?
I'd just default `_role` to `None`, though it looks like other parts of the code uses `hasattr` so removing all of those is maybe a separate PR - in which case you can simplify this PR by not changing the logic.

Reply:
Ideally you'd update the other hasattr's in this PR.